lib/openstax/aws/tag.rb

```diff
@@ -0,0 +1,27 @@
+module OpenStax::Aws
+  class Tag
+
+    attr_reader :key, :value
+
+    AWS_TAG_KEY_REGEX = /\A[\w\+\-\=\.\:\/\@]{1,128}\z/
+    AWS_TAG_VALUE_REGEX = /\A[\w\+\-\=\.\:\/\@]{0,256}\z/
+
+    def initialize(key, value)
+      if key.nil? || !key.match(AWS_TAG_KEY_REGEX)
+        raise "The tag key '#{key}' is invalid: must be a non-blank ID matching #{AWS_TAG_KEY_REGEX}"
+      end
+
+      @key = key.to_s
+
+      if @key.starts_with?("aws:")
```

Comment:
could do a negative lookahead and a ! expression to include this check in the regex match

Reply:
I try to stay positive.